docs/aws/audit/redshiftmonitoring/rules/emr_cluster_logging.mdx

```diff
@@ -23,6 +23,232 @@ HIPAA
+   ```sh
+   aws emr create-cluster --name "YourClusterName" --release-label emr-5.30.0 --applications Name=Hadoop Name=Spark --ec2-attributes KeyName=YourKeyName --instance-type m5.xlarge --instance-count 3 --log-uri s3://your-log-bucket-name/logs/
+   ```
+
+3. **Verify Logging Configuration:**
+   After creating the cluster, verify that logging is enabled by describing the cluster and checking the `LogUri` field:
+   ```sh
+   aws emr describe-cluster --cluster-id j-XXXXXXXXXXXXX
+   ```
+
+4. **Set Up IAM Role with S3 Access:**
+   Ensure the EMR cluster's IAM role has the necessary permissions to write logs to the S3 bucket. Attach the following policy to the IAM role:
+   ```sh
+   aws iam put-role-policy --role-name EMR_EC2_DefaultRole --policy-name S3AccessPolicy --policy-document '{
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Effect": "Allow",
+               "Action": [
+                   "s3:PutObject",
+                   "s3:PutObjectAcl"
+               ],
+               "Resource": "arn:aws:s3:::your-log-bucket-name/*"
+           }
+       ]
+   }'
+   ```
+
+By following these steps, you can ensure that EMR cluster logging is enabled and properly configured using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of EMR Cluster Logging not being enabled in AWS Redshift using Python scripts, you can follow these steps:
+
+1. **Install AWS SDK for Python (Boto3):**
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials:**
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by directly configuring the `~/.aws/credentials` file.
+
+3. **Create a Python Script to Enable EMR Cluster Logging:**
+   Write a Python script that will create an EMR cluster with logging enabled. Below is an example script:
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EMR
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   emr_client = session.client('emr')
+
+   # Define the EMR cluster configuration with logging enabled
+   cluster_config = {
+       'Name': 'MyEMRCluster',
+       'LogUri': 's3://my-emr-logs/',  # Ensure this S3 bucket exists and is accessible
+       'ReleaseLabel': 'emr-5.30.0',
+       'Instances': {
+           'InstanceGroups': [
+               {
+                   'Name': 'Master nodes',
+                   'Market': 'ON_DEMAND',
+                   'InstanceRole': 'MASTER',
+                   'InstanceType': 'm5.xlarge',
+                   'InstanceCount': 1,
+               },
+               {
+                   'Name': 'Core nodes',
+                   'Market': 'ON_DEMAND',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>